resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -237,7 +263,8 @@ private[spark] object KubernetesSparkRestServer {
       parsedArguments.host.get,
       parsedArguments.port.get,
       sparkConf,
-      secretBytes)
+      secretBytes,
+      barrier)
```

Comment:
The barrier needs to count down to unblock the main thread whenever we stop the server.